sql/api/src/main/scala/org/apache/spark/sql/internal/types/AbstractStringType.scala

```diff
@@ -23,41 +23,76 @@ import org.apache.spark.sql.types.{AbstractDataType, DataType, StringType}
 /**
  * AbstractStringType is an abstract class for StringType with collation support.
```

Comment:
let's update these comments to reflect the new state of things - adding trimming support